lib/core.ps1

```diff
@@ -576,6 +576,9 @@ function shim($path, $global, $name, $arg) {
         "if(`$myinvocation.expectingInput) { `$input | & `$path $arg @args } else { & `$path $arg @args }" | out-file "$shim.ps1" -encoding utf8 -append
     }
 
+    # make sure exitcode is returned from .ps1 shim
+    'exit $LastExitCode' | out-file "$shim.ps1" -encoding utf8 -append
```

Comment:
- I am not sure about the utf8 encoding in older powershell versions.. But since it is working now i guess it is okay
- lastexitCode is defined as `LASTEXITCODE`


```suggestion
    Add-Content "$shim.ps1" 'exit $LASTEXITCODE' -Encoding utf8
```